src/storage/mod.rs

```diff
@@ -117,10 +166,73 @@ impl Storage {
         }
     }
 
+    fn max_file_size_for(&self, path: &str) -> usize {
+        if path.ends_with(".html") {
+            self.max_file_size_html
+        } else {
+            self.max_file_size
+        }
+    }
+
+    pub(crate) fn fetch_rustdoc_file(
+        &self,
+        name: &str,
+        version: &str,
+        path: &str,
+        archive_storage: bool,
+    ) -> Result<Blob, Error> {
+        Ok(if archive_storage {
+            let archive = format!("rustdoc/{0}/rustdoc-{0}-{1}.zip", name, version);
+            self.get_from_archive(&archive, path, self.max_file_size_for(path))?
+        } else {
+            // Add rustdoc prefix, name and version to the path for accessing the file stored in the database
+            let remote_path = format!("rustdoc/{}/{}/{}", name, version, path);
+            self.get(&remote_path, self.max_file_size_for(path))?
+        })
+    }
+
+    pub(crate) fn fetch_source_file(
+        &self,
+        name: &str,
+        version: &str,
+        path: &str,
+        archive_storage: bool,
+    ) -> Result<Blob, Error> {
+        Ok(if archive_storage {
+            let archive = format!("sources/{0}/sources-{0}-{1}.zip", name, version);
+            self.get_from_archive(&archive, path, self.max_file_size_for(path))?
+        } else {
+            let remote_path = format!("sources/{}/{}/{}", name, version, path);
+            self.get(&remote_path, self.max_file_size_for(path))?
+        })
+    }
+
+    pub(crate) fn rustdoc_file_exists(
+        &self,
+        name: &str,
+        version: &str,
+        path: &str,
+        archive_storage: bool,
+    ) -> Result<bool, Error> {
+        Ok(if archive_storage {
+            let archive = format!("rustdoc/{0}/rustdoc-{0}-{1}.zip", name, version);
```

Comment:
Hmm, this is getting to be quite a bit of duplication - maybe you could add a `rustdoc_file_path` function that you can use everywhere instead of hardcoding this format string?

Reply:
I'll try to think of something